Update Cloudflare deploy/integration docs to mention Workers Assets (#11344)
withastro/docs: src/content/docs/en/guides/integrations-guide/cloudflare.mdx

## Cloudflare runtime

### Usage

The Cloudflare runtime gives you access to environment variables and bindings to Cloudflare resources.

> The Cloudflare runtime uses bindings found in the `wrangler.toml`/`wranger.json` configuration file. 

Reviewer: ```suggestion
The Cloudflare runtime uses bindings found in the `wrangler.toml`/`wrangler.json` configuration file. 
```

> You can access the bindings by using Astro locals like this:

Reviewer 2: ```suggestion
You can access the bindings from `Astro.locals.runtime`:
```
While I'm here anyway... 😄

}
```

See the [list of all supported bindings](https://developers.cloudflare.com/workers/wrangler/api/#supported-bindings) in the Cloudflare documentation.

### Environment variables and secrets

The Cloudflare runtime treats environment variables as a type of binding.

For example, you can define an [environment variable](https://developers.cloudflare.com/workers/configuration/environment-variables/#add-environment-variables-via-wrangler) in `wrangler.json` as follows:

```json title="wrangler.json"
{
  "vars" : {
    "MY_VARIABLE": "test"
  }
}
```

> Secrets are a special type of environment variable that allow you to attach encrypted text values to your Worker. They need to be defined differently to ensure they are not visible after you set them.

Reviewer 2: ❤️

> To define `secrets`, add them through the [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/) rather than in your Wrangler config file. 

Reviewer 2: ```suggestion

To define `secrets`, add them through the [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/) rather than in your Wrangler config file. 
```
Let's make this a new paragraph for readability!

```bash
npx wrangler secret put <KEY>
```

To set secrets for local development, you also need to add a `.dev.vars` file to the root of the Astro project:

```ini title=".dev.vars"
DB_PASSWORD=myPassword
```

You can then access environment variables, including secrets, from the `env` object available from `Astro.locals.runtime`:  

```astro title="src/pages/index.astro"
---
const { env } = Astro.locals.runtime;
const myVariable = env.MY_VARIABLE;
const secret = env.DB_PASSWORD;
---
```

> Cloudflare environment variables and secrets are compatible with the [`astro:env` API](/en/guides/environment-variables/#type-safe-environment-variables).

Reviewer 2: ❤️

}
```

See the [list of all supported bindings](https://developers.cloudflare.com/workers/wrangler/api/#supported-bindings) in the Cloudflare documentation.

> ### Environment variables and secrets

Reviewer 3: Can we specify somewhere that env vars/secrets can be used with `astro:env` too?

---
```

> However, to define `secrets`, you should add them through the [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/). 

Author: Reorganising is a great suggestion, I've done that now :)

You can attach [custom headers](https://developers.cloudflare.com/pages/platform/headers/) to your responses by adding a `_headers` file in your Astro project's `public/` folder. This file will be copied to your build output directory.

This is available on Cloudflare Workers and Pages.

### Assets
Assets built by Astro are all named with a hash and therefore can be given long cache headers. By default, Astro on Cloudflare will add such a header for these files.

### Redirects

You can declare [custom redirects](https://developers.cloudflare.com/pages/platform/redirects/) to redirect requests to a different URL. You can add a `_redirects` file in your Astro project's `public/` folder. This file will be copied to your build output directory.

This is available on Cloudflare Workers and Pages.

### Routes
#### Routing on Cloudflare Workers

For Cloudflare Workers, you do not need a `_routes.json` file. routing for static assets is based on the file structure in the build directory (e.g. `./dist`). If no match is found, this will fall back to the Worker for SSR.

> <ReadMore> Read more about [static asset routing with Cloudflare Workers](https://developers.cloudflare.com/workers/static-assets/routing/).</ReadMore>

Reviewer 2: Just putting a pin in this for now because we *usually* reserve this component for internal links within our own docs. We haven't formally established this as a rule, but we do it often enough that I want to make sure we discuss it and make a conscious decision!

Author: Totally fair! I've moved it just a regular text + link, happy to follow whatever guidance you land on :)

You can attach [custom headers](https://developers.cloudflare.com/pages/platform/headers/) to your responses by adding a `_headers` file in your Astro project's `public/` folder. This file will be copied to your build output directory.

This is available on Cloudflare Workers and Pages.

### Assets
Assets built by Astro are all named with a hash and therefore can be given long cache headers. By default, Astro on Cloudflare will add such a header for these files.

### Redirects

> You can declare [custom redirects](https://developers.cloudflare.com/pages/platform/redirects/) to redirect requests to a different URL. You can add a `_redirects` file in your Astro project's `public/` folder. This file will be copied to your build output directory.

Reviewer 2: ```suggestion
You can declare [custom redirects](https://developers.cloudflare.com/pages/platform/redirects/) to redirect requests to a different URL. To do so, add a `_redirects` file in your Astro project's `public/` folder. This file will be copied to your build output directory.
```

This is available on Cloudflare Workers and Pages.

### Routes
#### Routing on Cloudflare Workers

Routing for static assets is based on the file structure in the build directory (e.g. `./dist`). If no match is found, this will fall back to the Worker for SSR. Read more about [static asset routing with Cloudflare Workers](https://developers.cloudflare.com/workers/static-assets/routing/).

> Unlike [Cloudflare Pages](#routing-on-cloudflare-pages), with Workers, you do not need a `_routes.json` file. 

Reviewer 2: ```suggestion

Unlike [Cloudflare Pages](#routing-on-cloudflare-pages), with Workers, you do not need a `_routes.json` file. 
```

#### Routing on Cloudflare Pages

> For Cloudflare Pages, [routing](https://developers.cloudflare.com/pages/platform/functions/routing/#functions-invocation-routes) uses a `_routes.json` file to determine which requests are routed to the SSR function and which are served as static assets. By default, a `_routes.json` file will be automatically generated for your project based on its files and configuration.

Reviewer 2: ```suggestion
For Cloudflare Pages, [routing](https://developers.cloudflare.com/pages/platform/functions/routing/#functions-invocation-routes) uses a `_routes.json` file to determine which requests are routed to the server function and which are served as static assets. By default, a `_routes.json` file will be automatically generated for your project based on its files and configuration.
```
